detectron2/evaluation/coco_evaluation.py

```diff
@@ -80,6 +80,13 @@ def _tasks_from_config(self, cfg):
         if cfg.MODEL.MASK_ON:
             tasks = tasks + ("segm",)
         if cfg.MODEL.KEYPOINT_ON:
+            assert (
+                len(cfg.TEST.KEYPOINT_OKS_SIGMAS) == cfg.MODEL.ROI_KEYPOINT_HEAD.NUM_KEYPOINTS
+                or cfg.MODEL.ROI_KEYPOINT_HEAD.NUM_KEYPOINTS == 17
+            ), "The length of cfg.TEST.KEYPOINT_OKS_SIGMAS (default: 17) must be equal to " \
+               "cfg.MODEL.ROI_KEYPOINT_HEAD.NUM_KEYPOINTS. For more information please refer to " \
+               "http://cocodataset.org/#keypoints-eval."
+
```

Comment:
This is assuming that `MODEL.ROI_KEYPOINT_HEAD.NUM_KEYPOINTS` is the number of keypoints entering the evaluation, which may not be true, e.g. when a user implements a different model.
A better place to check this is in `_evaluate_predictions_on_coco` to minimize dependencies on config items.